.github/workflows/gh_statistics_bot.yml

```diff
@@ -0,0 +1,450 @@
+                try {
+                  let existingFileSha = null;
+                  try {
+                    const existing = await github.rest.repos.getContent({
+                      owner: context.repo.owner,
+                      repo: context.repo.repo,
+                      path: chart.path,
+                      ref: branchName
+                    });
+                    existingFileSha = existing.data.sha;
+                  } catch (e) {
+                    console.log(`File ${chart.path} doesn't exist on branch, creating new file`);
+                  }
+                  
+                  const params = {
+                    owner: context.repo.owner,
+                    repo: context.repo.repo,
+                    path: chart.path,
+                    message: `Update ${chart.path}`,
+                    content: Buffer.from(chart.content).toString('base64'),
+                    branch: branchName
+                  };
+                  
+                  if (existingFileSha) params.sha = existingFileSha;
+                  
+                  await github.rest.repos.createOrUpdateFileContents(params);
+                  console.log(`Updated on branch: ${chart.path}`);
+                } catch (error) {
+                  console.log(`Error updating ${chart.path} on branch: ${error.message}`);
+                }
+              }
+              
+              const latestMonth = monthlyData[monthlyData.length - 1];
+              return {
+                branchName: branchName,
+                message: `Charts updated on branch ${branchName} for ${latestMonth.month}`,
+                branchUrl: `https://github.com/${context.repo.owner}/${context.repo.repo}/tree/${branchName}`
+              };
+            }
+
+            try {
+              console.log('Starting monthly report generation...');
+              
+              const monthlyData = [];
+              const now = new Date();
+              
+              for (let i = 12; i >= 1; i--) {
+                const monthStart = new Date(now.getFullYear(), now.getMonth() - i, 1);
+                const monthEnd = new Date(now.getFullYear(), now.getMonth() - i + 1, 0);
+                
+                console.log(`Month ${i}: ${monthStart.toLocaleDateString()} to ${monthEnd.toLocaleDateString()}`);
+                
+                const since = monthStart.toISOString();
+                const until = monthEnd.toISOString();
+                const monthName = monthStart.toLocaleString('default', { month: 'short', year: 'numeric' });
+                
+                console.log(`Processing ${monthName} (${since} to ${until})...`);
+                
+                const [stats, prStats, issueStats, releaseStats] = await Promise.all([
+                  getSimpleMonthStats(since, until),
+                  getPRStats(since, until),
+                  getIssueStats(since, until),
+                  getReleaseStats(since, until)
+                ]);
```

Comment:
If i understand this correctly, we load the entire history here?
How long does this take?